load_data.py

```diff
@@ -0,0 +1,115 @@
+from datasets import get_dataset_split_names, load_dataset
+from os import environ
+from tokenizers import Tokenizer
+from tokenizers.models import BPE, Unigram, WordLevel, WordPiece
+from tokenizers.normalizers import NFD, Sequence, Strip, StripAccents
+from tokenizers.trainers import BpeTrainer, UnigramTrainer, WordLevelTrainer, WordPieceTrainer
+from torch.utils.data import DataLoader
+from transformers import PreTrainedTokenizerFast
+
+# Disable parallelism to avoid errors with DataLoaders later on
+environ["TOKENIZERS_PARALLELISM"] = "false"
+
+def get_loaders_tokenizer(dataset_name: str,
+                          tokenizer_name: str,
+                          seq_len: int,
+                          batch_size: int,
+                          vocab_size: int,
+                          data_dir: str,
+                          dataset_config: str=None,
+                          max_token_len: int=20) -> tuple[DataLoader, DataLoader, DataLoader, Tokenizer]:
+    """ Loads the WikiText2 dataset and returns DataLoaders.
+    Args:
+        seq_len (int): Context window/sequence length.
+        batch_size (int): Batch size.
+        vocab_size (int): Maximum vocabulary size.
+
+    Returns:
+        Tuple with the format: (Training DataLoader, Validation DataLoader,
+        Testing DataLoader, Tokenizer object).
+    """
+    assert ["test", "train", "validation"] == sorted(get_dataset_split_names(path=dataset_name, config_name=dataset_config)), \
+        f"The dataset {dataset_name} doesn't have a train, validation, and test split!"
+
+    # Retrieve iterators for each split of the dataset
+    entire_dataset = load_dataset(path=dataset_name, name=dataset_config, cache_dir=data_dir)
+
+    # Function to filter out undesired inputs. In this case, filter out instances with only whitespace
+    filter_fun = lambda instance_dict : bool(instance_dict["text"].strip())
+
+    # Filter out undesired data instances
+    entire_dataset = entire_dataset.filter(filter_fun)
+
+    # Create one of the four tokenizer models supported by Hugging Face
+    if tokenizer_name == "BPE":
+        tokenizer = Tokenizer(BPE(unk_token="<unk>"))
+        trainer = BpeTrainer(vocab_size=vocab_size,
+                             show_progress=True,
+                             special_tokens=["<pad>", "<bos>", "<eos>", "<unk>"],
```

Comment:
I think it would be better to have a special_tokens variable as a list above called something like special_tokens, and simply passing it to minimize code duplication and allow for easy alterations.